### PR TITLE
Resolve `Hip17UnhappyTokensSuite` errors

### LINKED ISSUE #8557: Resolve `Hip17UnhappyTokensSuite` errors
### Problem

We have the following tests suite `Hip17UnhappyTokensSuite` that has problems when it's running against modularization codebase.

### Solution

Resolve issues with the test suite. When ready add @hapitest annotations to the passing tests.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:
This PR fixes all, but one tests in the `Hip17UnhappyTokensSuite` by:
 - Adding necessary checks for token deletion in `TokenFeeScheduleUpdateHandler`, `TokenFreezeAccountHandler`, `TokenMintHandler` and `TokenUnfreezeAccountHandler`

⚠️ Added 2 new lines in one of the tests, which overrides the configuration for the auto renew duration, as the test create tokens with autoRenewDuration == 100. In most configurations this will fails, as the minimum duration is much higher. This change is mostly to avoid this test failing on most configurations.

**Related issue(s)**:

Fixes #8557 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUnfreezeAccountHandler.java [hunk 1]
```diff
@@ -127,6 +131,8 @@ private TokenRelation validateSemantics(
         final var tokenRel = tokenRelStore.getForModify(accountId, tokenId);
         validateTrue(tokenRel != null, TOKEN_NOT_ASSOCIATED_TO_ACCOUNT);
 
+        // Check that token is not deleted
+        validateFalse(token != null && token.deleted(), TOKEN_WAS_DELETED);
```
Comment (jsync-swirlds) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUnfreezeAccountHandler.java:135]: This may create another issue by combining two checks.
A possible improvement:
```suggestion
        validateFalse(token == null, /*Fill in appropriate code here*/);
        validateFalse(token.deleted(), TOKEN_WAS_DELETED);
```

Comment (georgi-l95) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUnfreezeAccountHandler.java]: Agree, added `if` because I'm not sure if there is appropriate response code for non existing token.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFeeScheduleUpdateHandler.java [hunk 2]
```diff
@@ -130,6 +132,7 @@ private Token validateSemantics(
         validateTrue(token != null, INVALID_TOKEN_ID);
         validateTrue(token.hasFeeScheduleKey(), TOKEN_HAS_NO_FEE_SCHEDULE_KEY);
         validateTrue(op.customFees().size() <= config.maxCustomFeesAllowed(), CUSTOM_FEES_LIST_TOO_LONG);
+        validateFalse(token.deleted(), TOKEN_WAS_DELETED);
```
[the remark this answers is not in the document]
Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFeeScheduleUpdateHandler.java]: Fixed

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java [hunk 3]
```diff
@@ -97,6 +97,10 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         // validate token exists
         final var token = tokenStore.get(tokenId);
         validateTrue(token != null, INVALID_TOKEN_ID);
+
+        // Check that token is not deleted
+        validateFalse(token.deleted(), TOKEN_WAS_DELETED);
+
```
[the remark this answers is not in the document]
Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java]: Fixed

**Review by mhess-swl — DISMISSED** [review]

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java [hunk 4]
```diff
@@ -314,6 +322,7 @@ private HapiSpec cannotUpdateNftFeeScheduleWhenDeleted() {
                 .given(
                         newKeyNamed(ADMIN_KEY),
                         newKeyNamed(FEE_SCHEDULE_KEY),
+                        newKeyNamed(SUPPLY_KEY),
```
Comment (rbair23) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java:325]: Is this change necessary? How was it passing before on mono-service but not now without it? Is it a bug in mono service?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java [hunk 5]
```diff
@@ -241,9 +245,12 @@ private HapiSpec cannotAssociateNftWhenDeleted() {
                 .then(tokenAssociate(ANOTHER_USER, NFTdeleted).hasKnownStatus(TOKEN_WAS_DELETED));
     }
 
+    @HapiTest
     public HapiSpec cannotUpdateNftWhenDeleted() {
         return defaultHapiSpec("cannotUpdateNftWhenDeleted")
                 .given(
+                        fileUpdate(APP_PROPERTIES)
+                                .overridingProps(AutoRenewConfigChoices.propsForAccountAutoRenewOnWith(1, 0L, 2, 2)),
```
Comment (rbair23) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java:253]: Is this change necessary? How was it passing before on mono-service but not now without it? Is it a bug in mono service?

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFeeScheduleUpdateHandler.java [hunk 6]
```diff
@@ -126,8 +127,7 @@ private Token validateSemantics(
             @NonNull final TokenFeeScheduleUpdateTransactionBody op,
             @NonNull final WritableTokenStore tokenStore,
             @NonNull final TokensConfig config) {
-        var token = tokenStore.get(op.tokenIdOrElse(TokenID.DEFAULT));
-        validateTrue(token != null, INVALID_TOKEN_ID);
+        var token = TokenHandlerHelper.getIfUsable(op.tokenIdOrElse(TokenID.DEFAULT), tokenStore);
```
Comment (rbair23) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFeeScheduleUpdateHandler.java:130]: I see three different approaches being used here:

```var token = TokenHandlerHelper.getIfUsable(op.tokenIdOrElse(TokenID.DEFAULT), tokenStore);```

```
final var token = tokenStore.get(tokenId);
validateFalse(token != null && token.deleted(), TOKEN_WAS_DELETED);
```

```
final var tokenId = context.body().tokenMintOrThrow().tokenOrThrow();
final var token = TokenHandlerHelper.getIfUsable(tokenId, tokenStore);
```

Is there a common way we should be doing this, or are each of these correct? I didn't dig into it further, it just struck me as odd that checking the token ID is being done differerently

Comment (georgi-l95) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFeeScheduleUpdateHandler.java:130]: You're right, i've changed some and tried to make them similar. I agree that we should be consistent.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java (hunk 5 above)
Comment (georgi-l95) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java:253]: Not 100% necessary, what this make is overrides the `bootstrap.properties` for `autoRenewDuration` as the test is trying to update a token with very low `autoRenewDuration`, which will fail in most cases only because by default in the `bootstrap.properties` file this is set to `ledger.autoRenewPeriod.minDuration=2592000`

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java (hunk 4 above)
Comment (georgi-l95) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java:325]: Yes, not sure how it was passing in the mono service. We are setting supply key to the token below, but if we don't create it here on line 325 it fails with Missing 'supply key', which makes sense.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFeeScheduleUpdateHandler.java (hunk 6 above)
Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFeeScheduleUpdateHandler.java:130]: Generally we want to use `getIfUsable(...)` because it checks for nonexistent, deleted, or paused tokens. However in some cases that's not appropriate, for example handling an unfreeze token call, so either we go back to checking a subset of those conditions manually, or we come up with various permutations of `getIfUsable(...)`. We've tried to only use `tokenStore.get(...)` in places where `getIfUsable` isn't correct, but I'm sure we've missed a spot or two

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUnfreezeAccountHandler.java [hunk 7]
```diff
@@ -130,10 +131,7 @@ private TokenRelation validateSemantics(
         validateTrue(tokenRel != null, TOKEN_NOT_ASSOCIATED_TO_ACCOUNT);
 
         // Check that the token itself exists and is valid
-        final var token = tokenStore.get(tokenId);
-        validateTrue(token != null, INVALID_TOKEN_ID);
-        validateFalse(token.deleted(), TOKEN_WAS_DELETED);
-
+        var token = TokenHandlerHelper.getIfUsable(tokenId, tokenStore);
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUnfreezeAccountHandler.java:132]: My mistake, I was thinking 'freeze' and 'pause' were the same operation. `getIfUsable` should be the correct choice here (as it's written) 👍

**Review by mhess-swl — DISMISSED** [review]

**Comment by georgi-l95** [comment]
@mhess-swl some unit tests we failing because of the latest changes, i've fixed them and push, so approval was lost 😅

**Review by rbair23 — APPROVED** [review]

**Review by jsync-swirlds — APPROVED** [review]
